server/src/main/java/org/apache/druid/server/http/MetadataResource.java

```diff
@@ -180,26 +196,67 @@ private Response getAllUsedSegmentsWithAdditionalDetails(
           .filter(dataSourceWithUsedSegments -> dataSources.contains(dataSourceWithUsedSegments.getName()))
           .collect(Collectors.toList());
     }
-    final Stream<DataSegment> usedSegments = dataSourcesWithUsedSegments
-        .stream()
-        .flatMap(t -> t.getSegments().stream());
     final Set<DataSegment> overshadowedSegments = dataSourcesSnapshot.getOvershadowedSegments();
+    final Set<SegmentId> segmentAlreadySeen = new HashSet<>();
+    final Stream<SegmentStatusInCluster> segmentStatus = dataSourcesWithUsedSegments
+        .stream()
+        .flatMap(t -> t.getSegments().stream())
+        .map(segment -> {
+          // The replication factor for unloaded segments is 0 as they will be unloaded soon
+          boolean isOvershadowed = overshadowedSegments.contains(segment);
+          Integer replicationFactor = isOvershadowed ? (Integer) 0
+                                                     : coordinator.getReplicationFactor(segment.getId());
+
+          Long numRows = null;
+          if (null != coordinatorSegmentMetadataCache) {
+            AvailableSegmentMetadata availableSegmentMetadata = coordinatorSegmentMetadataCache.getAvailableSegmentMetadata(
+                segment.getDataSource(),
+                segment.getId()
+            );
+            if (null != availableSegmentMetadata) {
+              numRows = availableSegmentMetadata.getNumRows();
+            }
+          }
+          segmentAlreadySeen.add(segment.getId());
+          return new SegmentStatusInCluster(
+              segment,
+              isOvershadowed,
+              replicationFactor,
+              numRows,
+              // published segment can't be realtime
+              false
+          );
+        });
+
+    Stream<SegmentStatusInCluster> finalSegments = segmentStatus;
 
-    final Stream<SegmentStatusInCluster> segmentStatus = usedSegments.map(segment -> {
-      // The replication factor for unloaded segments is 0 as they will be unloaded soon
-      boolean isOvershadowed = overshadowedSegments.contains(segment);
-      Integer replicationFactor = isOvershadowed ? (Integer) 0
-                                                 : coordinator.getReplicationFactor(segment.getId());
+    // conditionally add realtime segments information
+    if (null != includeRealtimeSegments && null != coordinatorSegmentMetadataCache) {
```

Comment:
Since the realtime flag only works if the `centralizedSchemaManagement`is enabled lets document this in user facing docs and well as in the code. 

In the code we should mention that we did not want to increase the payload of the broker to coordinator communication if this feature is disabled and we did not want to introduce a new feature flag on the broker so this was a trade off done. This api will be deprecated once we fully move over to coordinator to power sys.segments q's from the broker.